File: python/armstrong_numbers.py
# How can you make this more scalable and reusable later?
def find_armstrong_numbers(numbers):
    armstrongs = []
    
    for num in numbers:
        num_list = list(map(int, str(num)))
        power = len(num_list)
        sum = 0
        for digit in num_list:
            sum += digit ** power  
        if sum == num:
            armstrongs.append(num)
            
    return armstrongs       

File: js/armstrongNumbers.js
// How can you make this more scalable and reusable later?
exports.findArmstrongNumbers = function(numbers) {
    armstrongs = [];

    for(let num of numbers){
        const numStr = num.toString()
        const digits = numStr.split('').map(digit=> parseInt(digit))
        const power = digits.length

        let sum = 0
        for (let digit of digits){
            sum += Math.pow(digit, power)       
        }
         if (sum === num){
            armstrongs.push(num)
        }
    }
    return armstrongs
};
